app/presenters/filter_presenter.rb

```diff
@@ -47,29 +62,25 @@ def organisation_options
         @organisations.map do |org|
           name = org[:name]
           name.concat " (#{org[:acronym]})" if org[:acronym].present?
-          {
-            text: name,
-            value: org[:id],
-            selected: org[:id] == @search_parameters[:organisation_id]
-          }
+          [name, org[:id]]
         end
     end
   end
 
 private
 
   def find_selected_org
-    organisation_options.find { |o| o[:selected] }
+    organisation_options.select { |o| o[1] == @search_parameters[:organisation_id] }[0] || ['All organisations', 'all']
```

Comment:
Are we skipping the first option because it's blank? Can we skip it before we do the select? Also with `[0]` would `[:name]` not work? Apologies for not seeing it clearly. Same for the document type select.

Reply:
We've built an array of arrays:
[[name1, id1],[name2, id2]] 
because this is the format the component needs.
We're querying that to find the option with the right value, then returning that name and id as an array. The [0] is to get the contents of the container array, the o[1] is to query against the 2nd part of the array where the id is as it's not a hash at this point. Happy to change it if there's a clearer way to write it.

Reply:
I've confused myself, now. Revisiting this.

Reply:
Okay, so I was using select, which returned an array, so I've changed that and been able to lose the [0]. This needs to return a slice of the array (which is an array itself) so I couldn't use [:name] here.